dynamax/utils.py

```diff
@@ -87,4 +87,50 @@ def random_rotation(seed, n, theta=None):
     out = jnp.eye(n)
     out = out.at[:2, :2].set(rot)
     q = jnp.linalg.qr(jr.uniform(key2, shape=(n, n)))[0]
-    return q.dot(out).dot(q.T)
\ No newline at end of file
+    return q.dot(out).dot(q.T)
+
+
+def ensure_array_has_batch_dim(tree, instance_shapes):
+    """Add a batch dimension to a PyTree, if necessary.
+
+    Example: If `tree` is an array of shape (100, 2) and
+    `instance_shapes` is a tuple (2,), then the return
+    value is the array with an added batch dimension, with
+    shape (1, 100, 2).
```

Comment:
Maybe explain that this can be used if T=100 is the number of time steps, D=2 is the size of the observation vector, and N=1 is the number of sequences.

Reply:
What if T=1, so input is a single observation (2,) and instance shape is also (2,)? Is output (1,1,2) or (1,2)?

Reply:
Our convention is that the emissions for a single time series are shape `(T,) + self.emissions_shape`, so if `T=1` then the user should still include the leading dimension in the  emissions. The batched emissions would then be shape `(1,1,D)`. 

For LGSSM, the inputs are assumed to be shape `(T,D)` where `D` is the input dimension. So if `input_dim == 1`, you _would_ need the trailing dimension @xinglong-li .